### ISSUE TITLE
microk8s status says "not running" but microk8s inspect shows everything is running

### ISSUE THREAD

**Issue by kquinsland** [ubuntu/microk8s#2008]
I have a 7 node cluster. A few days ago, I noticed one of the pods I have schedule for execution only one a specific node was no longer functioning. I have not been able to get time to write up the issue until now, however.

I checked node status and i saw that 3 of the 7 nodes were `NotReady`. I hadn't noticed the prior failures because of spare nodes so i don't quite know the timeline for failures. In any event, all nodes had pending package updates so I did those and rebooted the nodes.

After all the dust settled, only ONE of the 7 nodes is still `NotReady`:

```shell
root@node04:/home/karl# kubectl get no
NAME     STATUS     ROLES    AGE   VERSION
node05   NotReady   <none>   92d   v1.20.1-34+e7db93d188d0d1
node04   Ready      <none>   92d   v1.20.1-34+e7db93d188d0d1
node02   Ready      <none>   31d   v1.20.1-34+e7db93d188d0d1
node07   Ready      <none>   29d   v1.20.1-34+e7db93d188d0d1
node06   Ready      <none>   54d   v1.20.1-34+e7db93d188d0d1
node01   Ready      <none>   92d   v1.20.1-34+e7db93d188d0d1
node03   Ready      <none>   29d   v1.20.1-34+e7db93d188d0d1
```

But when i check on the inspect from the troubled `node05`, I get this output:

```shell
root@node05:/home/karl# microk8s inspect
Inspecting Certificates
Inspecting services
  Service snap.microk8s.daemon-cluster-agent is running
  Service snap.microk8s.daemon-containerd is running
  Service snap.microk8s.daemon-apiserver is running
  Service snap.microk8s.daemon-apiserver-kicker is running
  Service snap.microk8s.daemon-control-plane-kicker is running
  Service snap.microk8s.daemon-proxy is running
  Service snap.microk8s.daemon-kubelet is running
  Service snap.microk8s.daemon-scheduler is running
  Service snap.microk8s.daemon-controller-manager is running
  Copy service arguments to the final report tarball
Inspecting AppArmor configuration
Gathering system information
  Copy processes list to the final report tarball
  Copy snap list to the final report tarball
  Copy VM name (or none) to the final report tarball
  Copy disk usage information to the final report tarball
  Copy memory usage information to the final report tarball
  Copy server uptime to the final report tarball
  Copy current linux distribution to the final report tarball
  Copy openSSL information to the final report tarball
  Copy network configuration to the final report tarball
Inspecting kubernetes cluster
  Inspect kubernetes cluster
Inspecting juju
  Inspect Juju
Inspecting kubeflow
  Inspect Kubeflow

Building the report tarball
  Report tarball is at /var/snap/microk8s/1910/inspection-report-20210213_185938.tar.gz
root@node05:/home/karl# microk8s status
microk8s is not running. Use microk8s inspect for a deeper inspection.
```

When i look through the inspection report, i see that the API server is thrashing:

```
Feb 13 18:58:44 node05 systemd[1]: snap.microk8s.daemon-apiserver.service: Scheduled restart job, restart counter is at 43468.
```

Inspection report attached. I'll wait for guidance here before I go through the usual "remove, stop, reset, re-join" dance

[inspection-report-20210213_185938.tar.gz](https://github.com/ubuntu/microk8s/files/5976291/inspection-report-20210213_185938.tar.gz)

**Comment by balchua** [comment]
@kquinsland there are some fixes made recently on dqlite (which is the datastore used by MicroK8s for HA cluster).
You may want to check the comment below.

https://github.com/ubuntu/microk8s/issues/1598#issuecomment-779044844

**Comment by kquinsland** [comment]
@balchua Thanks!  I saw that / am subscribed to that issue thread.  I am looking forward to the changes landing!

I posted this issue as a "FYI" of sorts. The `microk8s status` and `microk8s inspect` seem to be testing 'health' of the various services differently... which is why they lead to different results.  I wasn't sure if this was intentional or not.

running `microk8s inspect` shows the `Inspecting services...` output which appears to just check `systemctl is-running` which is very different from "is the service functional?"


I'll force remove `node05` from the cluster and wait until the backports land in the stable/latest channels before re-adding the node.
